website/static/js/fangorn.js

```diff
@@ -1609,14 +1613,37 @@ function expandStateLoad(item) {
             }
         }
     }
+
     if (item.children.length > 0 && item.depth === 2) {
         for (i = 0; i < item.children.length; i++) {
             if (item.children[i].data.isAddonRoot || item.children[i].data.addonFullName === 'OSF Storage' ) {
                 tb.updateFolder(null, item.children[i]);
             }
         }
     }
-        $('.fangorn-toolbar-icon').tooltip();
+
+    if (item.depth > 2 && !item.data.isAddonRoot && item.children.length === 0 && item.open) {
+        if (icon.get(0)) {
+            m.render(icon.get(0), tbOptions.resolveRefreshIcon());
```

Comment:
Calling `m.render` is generally a bad idea. Can you explain why it's necessary here?

Reply:
This renders the loading indicator until the request a few lines below finishes. It's exactly what treebeard does ([see here](https://github.com/CenterForOpenScience/treebeard/blob/75eca74fc66009e42bb5629481858939524701e0/scripts/grid.js#L1044)), I'm just avoiding the rest of the `toggleFolder` functionality. 

However, the second `m.render` (in the `.done()`) isn't necessary so I'm removing that.

Reply:
Ah, ok, thanks. It would be worth adding a comment that this is taken from treebeard.

Reply:
Good point. I'll add that real quick.